src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/modelchecking/ModelCheckingStrategy.kt

```diff
@@ -103,7 +103,9 @@ internal class ModelCheckingStrategy(
         // as the instrumented code might call `enterIgnoredSection` just before this call.
         return replay && collectTrace &&
                 Thread.currentThread() is TestThread &&
-                suddenInvocationResult == null
+                suddenInvocationResult == null &&
+                !shouldSkipNextBeforeEvent()
```

Comment:
Why would you need to skip the next beforeEvent? Why should it be added now?

Reply:
This is essentially a hack to make the plugin integration working without refactoring too much code.

So the problem is because of the coroutine resumptions. 
We don't have coroutine resumptions events as trace points in the trace. Thus they don't have "eventId"-s which are used to integrate Lincheck with the plugin. 

However, when the resumption is called, a lot of stuff is happening under the hood. 
In particular, a `suspend fun` is called with the given completion object.
`MethodCallTransformer` instruments this call as it instruments other method calls, however, in case of resumption this `suspend fun` call should not create new trace point. 
`MethodCallTransformer` does not know about it, it always injects `beforeEvent` call regardless:

```
invokeStatic(Injections::beforeMethodCall)
invokeBeforeEventIfPluginEnabled("method call $methodName", setMethodEventId = true)  
```

Therefore, to "skip" this `beforeEvent` call following resumption call to `suspend fun`, we use this `skipNextBeforeEvent` flag. 


IMO, a better approach would be to refactor a code, and instead just assign `eventId`-s directly to trace points. 
Methods like `beforeMethodCall` then can return `eventId` of the created trace point, or something like `-1` in case when no trace point is created. Then subsequent `beforeEvent` call can just take this `eventId` from the stack. 

But this would be quite a large refactoring, I think it's better to implement it as separate PR. 

@avpotapov00 WDYT?

Reply:
I'm OK with this hack, please document it in the source code.

Reply:
Agreed to leave this for now. Should add a comment to source code.